src/main/java/org/sagebionetworks/web/client/presenter/EntityPresenter.java

```diff
@@ -401,24 +403,23 @@ public Widget asWidget() {
 
   @EventHandler
   public void onEntityUpdatedEvent(EntityUpdatedEvent event) {
-    List<SynapseReactClientEntityQueryKey> queryKey =
-      SynapseReactClientEntityQueryKey.create(
-        QueryKeyConstants.ENTITY,
-        event.getEntityId()
-      );
-    queryClient.resetQueries(queryKey);
+    KeyFactory keyFactory = keyFactoryProvider.getKeyFactory(
+      authenticationController.getCurrentUserAccessToken()
+    );
+
+    List<?> queryKey = keyFactory.getEntityQueryKey(event.getEntityId());
+    queryClient.invalidateQueries(queryKey);
```

Comment:
Use the tool to generate the cache key instead of relying on a consistent implementation